### PR TITLE
feat: Resize user avatars depedns on the needed display size - EXO-60290 - Meeds-io/MIPs#25

### PR DESCRIPTION
**Pull request by hakermi**
Prior to this change, avatars are displayed and fetched with their original size, which affects the perf and leading to a critical loading time in the stream page because of the number of the enormous loaded images.
This PR:
1 - Use the ResizeImageService and applying a resize step on avatars (users and spaces) on server side before rendering it on the page which enhance the perf on the stream page and the images load time. The resized image in store by using Metadata Service
2 - Make sure to drop related metadata when the avatar was updated to recreate it when resize the new uploaded image.

### CODE REVIEW

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you rework the PR description ? You haev 3 commits description currently

**Comment by hakermi** [comment]
> Can you rework the PR description ? You haev 3 commits description currently

Done

File: component/api/src/main/java/org/exoplatform/social/metadata/thumbnail/ImageThumbnailService.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.exoplatform.social.metadata.thumbnail;
```
Comment (rdenarie) [inline component/api/src/main/java/org/exoplatform/social/metadata/thumbnail/ImageThumbnailService.java:16]: Header is missing

File: component/api/src/main/java/org/exoplatform/social/metadata/thumbnail/model/ThumbnailObject.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.exoplatform.social.metadata.thumbnail.model;
```
Comment (rdenarie) [inline component/api/src/main/java/org/exoplatform/social/metadata/thumbnail/model/ThumbnailObject.java:16]: Header is missing

File: component/core/src/main/java/org/exoplatform/social/core/metadata/thumbnail/ImageThumbnailServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
+package org.exoplatform.social.core.metadata.thumbnail;
```
Comment (rdenarie) [inline component/core/src/main/java/org/exoplatform/social/core/metadata/thumbnail/ImageThumbnailServiceImpl.java:16]: Header is missing

File: component/core/src/test/java/org/exoplatform/social/core/metadata/thumbnail/ImageThumbnailServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
+package org.exoplatform.social.core.metadata.thumbnail;
```
Comment (rdenarie) [inline component/core/src/test/java/org/exoplatform/social/core/metadata/thumbnail/ImageThumbnailServiceImplTest.java:16]: Header is missing

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: component/api/src/main/java/org/exoplatform/social/metadata/thumbnail/ImageThumbnailService.java (hunk 1 above)
Comment (hakermi) [reply component/api/src/main/java/org/exoplatform/social/metadata/thumbnail/ImageThumbnailService.java:16]: done

File: component/api/src/main/java/org/exoplatform/social/metadata/thumbnail/model/ThumbnailObject.java (hunk 2 above)
Comment (hakermi) [reply component/api/src/main/java/org/exoplatform/social/metadata/thumbnail/model/ThumbnailObject.java:16]: done

File: component/core/src/main/java/org/exoplatform/social/core/metadata/thumbnail/ImageThumbnailServiceImpl.java (hunk 3 above)
Comment (hakermi) [reply component/core/src/main/java/org/exoplatform/social/core/metadata/thumbnail/ImageThumbnailServiceImpl.java:16]: done

File: component/core/src/test/java/org/exoplatform/social/core/metadata/thumbnail/ImageThumbnailServiceImplTest.java (hunk 4 above)
Comment (hakermi) [reply component/core/src/test/java/org/exoplatform/social/core/metadata/thumbnail/ImageThumbnailServiceImplTest.java:16]: done

**Review by rdenarie — APPROVED** [review]

**Pull request merged**